astroplayer-core/native/AstroPlayer/AstroPlayer.swift

```diff
@@ -0,0 +1,159 @@
+import Combine
+import Foundation
+import AVFoundation
+
+@objc public protocol AstroListener { }
+@objc public protocol AstroMediaItem  { }
+@objc public protocol EqualizerValues { }
+
+@objc public enum RepeatMode: Int {
+    case off
+    case all
+    case one
+}
+
+@objc public class iOSAstroPlayer : NSObject {
+    let player = AVQueuePlayer(items: [])
+
+    @objc public var mediaItemCount: Int32 {
+        Int32(player.items().count)
+    }
+
+    @objc public var contentDuration: Int64 = 0
+    @objc public var currentEqualizerValues: EqualizerValues? = nil
+    @objc public var currentMediaItem: AstroMediaItem? = nil
+    @objc public var currentMediaItemIndex: Int32 = 0
+    @objc public var currentPosition: Int64 = 0
+
+    @objc public var isEqualizerEnabled: Bool = false
+    @objc public var isMuted: Bool = false
+    @objc public var isPaused: Bool = false
+    @objc public var isPlaying: Bool = false
+    @objc public var isSmartEqualizerEnabled: Bool = false
+
+    @objc public var volume: Float = 0.0
+
+    @objc public var playBackSpeed: Float = 0.0
+    @objc public var repeatMode: RepeatMode = .off
+
+    @objc public var seekBackIncrement: Int64 = 0
+    @objc public var seekForwardIncrement: Int64 = 0
+    @objc public var shuffleModeEnabled: Bool = false
+    @objc public var smartEqualizerPicker: ((String) -> EqualizerValues)?
+
+    // MARK: -
+
+    @objc public func prepare() {
+        debugPrint(#function)
+    }
+
+    // MARK: -
+
+    @objc public func play() {
+        debugPrint(#function)
+    }
+
+    @objc public func pause() {
+        debugPrint(#function)
+    }
+
+    @objc public func seekTo(milliseconds: Int64) {
+        debugPrint(#function)
+    }
+
+    @objc public func seekToMediaItem(index: Int32) {
+        debugPrint(#function)
+    }
+
+    // MARK: -
+
+    @objc public func registerNativeListenerForAstro() {
+        debugPrint(#function)
+    }
+
+    @objc public func deregisterNativeListenerForAstro() {
+        debugPrint(#function)
+    }
+
+    // MARK: -
+
+    @objc public func mute() {
+        debugPrint(#function)
+    }
+
+    @objc public func unMute() {
+        debugPrint(#function)
+    }
+
+    @objc public func increaseVolume() {
+        debugPrint(#function)
+    }
+
+    @objc public func decreaseVolume() {
+        debugPrint(#function)
+    }
+
+    @objc public func decreaseVolumeBy(offset: Float) {
+        debugPrint(#function)
+    }
+
+    @objc public func increaseVolumeBy(offset: Float) {
+        debugPrint(#function)
+    }
+
+    // MARK: -
+
+    @objc public func addMediaItem(item: AstroMediaItem) {
+        debugPrint(#function)
+    }
+
+    @objc public func addMediaItem(index: Int32, item: AstroMediaItem) {
+        debugPrint(#function)
+    }
+
+    @objc public func addMediaItems(items: [AstroMediaItem]) {
```

Comment:
The generated KNM includes `WithItems`:

```kotlin
@kotlin.commonizer.ObjCCallable
public open expect fun addMediaItemsWithItems(items: kotlin.collections.List<*>): kotlin.Unit { /* compiled code */ }
```

which does not match the requirement for expect:

```kotlin
actual open fun addMediaItems(items: List<AstroMediaItem>) {
    // Implementation code
}
```

This discrepancy arises because the generated method uses `List<*>` to represent the Objective-C `NSArray`, whereas we need a specific type `List<AstroMediaItem>`.